ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -331,10 +360,16 @@ void helperMap(List<OntologyHierarchy> children) {
     @Override
     public void onSaveOntologyHierarchy(SaveOntologyHierarchyEvent event) {
         List<String> iris = event.getIris();
-
-        for (final String iri : iris) {
+        boolean clearHierarchies = true;
+        boolean isLast = false;
+        for (String iri : iris) {
+            if (iris.indexOf(iri) == iris.size() - 1) {
+                isLast = true;
+            }
+            view.maskHierarchyTree();
             serviceFacade.saveOntologyHierarchy(event.getOntology().getVersion(),
-                                                iri, new SaveHierarchyAsyncCallback(iri));
+                                                iri, new SaveHierarchyAsyncCallback(iri, clearHierarchies, isLast));
+            clearHierarchies = false;
```

Comment:
There is no guarantee that the first async call will return success first, or that the last async call will return success last.
So I don't think we can rely on these boolean flags for clearing and refreshing the hierarchy listing.
If we still want to support adding multiple hierarchies at a time, the easiest solution might be to just refresh the hierarchies from the service after each `onSuccess` returns.